src/m3/functions.ts

```diff
@@ -26,7 +28,17 @@ import {flatmapNonCyclingFollowing} from "../utils/recursion.ts"
  * @return The type of the given {@link Feature}
  */
 export const type = (feature: Feature): FeaturesContainer | Datatype | typeof unresolved =>
-    (feature as Link | Property).type
+    (feature as (Link | Property)).type
+
+
+export const isRealProperty = (feature: Feature): feature is Property =>
+    feature instanceof Property && !feature.derived
+
+export const isRealContainment = (feature: Feature): feature is Containment =>
```

Comment:
Same here

Reply:
> +1 I cannot find anything obviously wrong, but I am not enough skilled in TS to provide valuable feedback

That makes it all the more important that the code is readable and understandable for non-TS experts as well.